File: solutions/python/N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n==0:
            return 0;
        fibos=[0,1,1]
        while len(fibos)<=n:
            fibos.append(fibos[len(fibos)-1]+fibos[len(fibos)-2]+fibos[len(fibos)-3])
        return fibos[len(fibos)-1]

File: solutions/cpp/N-th Tribonacci Number.cpp
#include <vector>
class Solution
{
public:
    int tribonacci(int n)
    {
        if (n < 1)
            return 0;
        std::vector<int> nums = {0, 1, 1};
        while (nums.size() <= n)
        {
            nums.push_back(nums[nums.size() - 1] + nums[nums.size() - 2] + nums[nums.size() - 3]);
        }
        return nums[nums.size() - 1];
    }
};